src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderProfileDetail.java

```diff
@@ -61,6 +61,13 @@ public static class PK implements Serializable {
     @NotFound(action = IGNORE)
     private ProfileCode code;
 
+    public void setProfileCode(ProfileCode code) {
+        this.profileCode = code.getId().getCode();
+    }
+
+    @Column(name = "PROFILE_CODE")
+    private String profileCode;
+
```

Comment:
ProfileCode is immutable due to mapping (update bale false etc), this is due to ProfileType mapped as primary key above.  So double map the actual code so it is updatable - hence need for strange `setProfileCode` helper